src/lib/CMakeLists.txt

```diff
@@ -5,11 +5,14 @@ if(BUILD_LIBRARY)
 	return()
 endif()
 
+if(NOT CONFIG_AGENT_DISABLE)
+	add_local_sources(sof agent.c)
```

Comment:
Can we name this CONFIG_HAVE_AGENT. I don't like double negations. Its hard to mentally process this :). Other than that I'm fine with the change.